src/features/projectsV2/ProjectsContext.tsx

```diff
@@ -5,20 +5,48 @@ import {
   useEffect,
   useMemo,
   useState,
+  useCallback,
 } from 'react';
 import { MapProject } from '../common/types/projectv2';
 import { useLocale } from 'next-intl';
 import getStoredCurrency from '../../utils/countryCurrency/getStoredCurrency';
 import { getRequest } from '../../utils/apiRequests/api';
 import { ErrorHandlingContext } from '../common/Layout/ErrorHandlingContext';
-import { APIError, handleError } from '@planet-sdk/common';
+import {
+  APIError,
+  handleError,
+  TreeProjectClassification,
+} from '@planet-sdk/common';
 import { useTenant } from '../common/Layout/TenantContext';
 import { SetState } from '../common/types/common';
+import { getSearchProjects } from './ProjectListControls/utils';
+import { ProjectTabs } from './ProjectListControls';
+import { doesProjectExistInFilteredLists } from './ProjectListControls/utils';
+
+const MOBILE_BREAKPOINT = 481;
+const TAB_OPTIONS = {
+  TOP_PROJECTS: 'topProjects',
+  ALL_PROJECTS: 'allProjects',
+} as const;
 
 interface ProjectsState {
   projects: MapProject[] | null;
   isLoading: boolean;
   isError: boolean;
+  filteredTopProjects: MapProject[] | null;
+  setFilteredTopProjects: SetState<MapProject[] | null>;
+  filteredRegularProjects: MapProject[] | null;
+  setFilteredRegularProjects: SetState<MapProject[] | null>;
```

Comment:
Rename this as `filteredAllProjects` since `regularProjects` is used to describe the type of project (top, regular, non donatable) elsewhere.